urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/SevenZip/LzmaAlone.java

```diff
@@ -0,0 +1,331 @@
+                        {
+                            return false;
+                        }
+                    }
+                    else
+                    {
+                        InFile = s;
+                    }
+                }
+                else if (pos == 2)
+                {
+                    OutFile = s;
+                }
+                else
+                {
+                    return false;
+                }
+                pos++;
+                continue;
+            }
+            return true;
+        }
+    }
+    
+    static void PrintHelp()
+    {
+        System.out
+                .println("\nUsage:  LZMA <e|d> [<switches>...] inputFile outputFile\n"
+                        + "  e: encode file\n"
+                        + "  d: decode file\n"
+                        + "  b: Benchmark\n"
+                        + "<Switches>\n"
+                        +
+                        // "  -a{N}:  set compression mode - [0, 1], default: 1 (max)\n"
+                        // +
+                        "  -d{N}:  set dictionary - [0,28], default: 23 (8MB)\n"
+                        + "  -fb{N}: set number of fast bytes - [5, 273], default: 128\n"
+                        + "  -lc{N}: set number of literal context bits - [0, 8], default: 3\n"
+                        + "  -lp{N}: set number of literal pos bits - [0, 4], default: 0\n"
+                        + "  -pb{N}: set number of pos bits - [0, 4], default: 2\n"
+                        + "  -mf{MF_ID}: set Match Finder: [bt2, bt4], default: bt4\n"
+                        + "  -eos:   write End Of Stream marker\n");
+    }
+    
+    public static void main(String[] args) throws Exception
+    {
+        System.out.println("\nLZMA (Java) 4.61  2008-11-23\n");
+        
+        if (args.length < 1)
+        {
+            PrintHelp();
+            return;
+        }
+        
+        final CommandLine params = new CommandLine();
+        if (!params.Parse(args))
+        {
+            System.out.println("\nIncorrect command");
+            return;
+        }
+        
+        if (params.Command == CommandLine.kBenchmak)
+        {
+            int dictionary = 1 << 21;
+            if (params.DictionarySizeIsDefined)
+            {
+                dictionary = params.DictionarySize;
+            }
+            if (params.MatchFinder > 1)
+            {
+                throw new Exception("Unsupported match finder");
+            }
+            SevenZip.LzmaBench.LzmaBenchmark(params.NumBenchmarkPasses,
+                    dictionary);
+        }
+        else if (params.Command == CommandLine.kEncode
+                || params.Command == CommandLine.kDecode)
+        {
+            final java.io.File inFile = new java.io.File(params.InFile);
+            final java.io.File outFile = new java.io.File(params.OutFile);
+            
+            final java.io.BufferedInputStream inStream = new java.io.BufferedInputStream(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *inStream*. The resource is referenced at lines 243, 291, 297, 309, 316. The resource is closed at line 323. There are other execution paths that do not contain closure statements. Ensure *inStream* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).